packages/sanity/src/core/versions/components/dialog/BundleForm.tsx

```diff
@@ -1,43 +1,58 @@
-/* eslint-disable no-warning-comments */
 /* eslint-disable i18next/no-literal-string */
 import {CalendarIcon} from '@sanity/icons'
-import {
-  Box,
-  Button,
-  type ButtonTone,
-  Card,
-  Flex,
-  Select,
-  Stack,
-  Text,
-  TextArea,
-  TextInput,
-} from '@sanity/ui'
-import {useCallback, useState} from 'react'
+import {Box, Button, Card, Flex, Popover, Stack, Text, TextArea, TextInput} from '@sanity/ui'
+import {useCallback, useMemo, useState} from 'react'
+import {useDateTimeFormat, useTranslation} from 'sanity'
 import speakingurl from 'speakingurl'
 
-import {type Bundle} from '../../types'
-import {RANDOM_TONES} from '../../util/const'
+import {type CalendarLabels} from '../../../form/inputs/DateInputs/base/calendar/types'
+import {DatePicker} from '../../../form/inputs/DateInputs/base/DatePicker'
+import {getCalendarLabels} from '../../../form/inputs/DateInputs/utils'
+import {type BundleDocument} from '../../../store/bundles/types'
 import {isDraftOrPublished} from '../../util/dummyGetters'
+import {BundleIconEditorPicker} from './BundleIconEditorPicker'
 
 export function BundleForm(props: {
-  onChange: (params: Bundle) => void
-  value: Bundle
+  onChange: (params: Partial<BundleDocument>) => void
+  value: Partial<BundleDocument>
 }): JSX.Element {
   const {onChange, value} = props
+  const {title, description, icon, hue, publishAt} = value
+
+  const dateFormatter = useDateTimeFormat()
+
   const [showTitleValidation, setShowTitleValidation] = useState(false)
+  const [showDateValidation, setShowDateValidation] = useState(false)
+  const [showDatePicker, setShowDatePicker] = useState(false)
+
+  const publishAtDisplayValue = useMemo(() => {
+    if (!publishAt) return ''
+    return dateFormatter.format(new Date(publishAt as Date))
+  }, [dateFormatter, publishAt])
+
+  const [displayDate, setDisplayDate] = useState(publishAtDisplayValue)
+  const {t: coreT} = useTranslation()
+  const calendarLabels: CalendarLabels = useMemo(() => getCalendarLabels(coreT), [coreT])
+
+  const iconValue: Partial<BundleDocument> = useMemo(
+    () => ({
+      icon: icon ?? 'cube',
+      hue: hue ?? 'gray',
+    }),
+    [icon, hue],
+  )
 
   const handleBundleTitleChange = useCallback(
     (event: React.ChangeEvent<HTMLInputElement>) => {
-      const title = event.target.value
+      const pickedTitle = event.target.value
 
-      if (isDraftOrPublished(title)) {
+      if (isDraftOrPublished(pickedTitle)) {
         setShowTitleValidation(true)
       } else {
         setShowTitleValidation(false)
       }
 
-      onChange({...value, title: title, name: speakingurl(title)})
+      onChange({...value, title: pickedTitle, name: speakingurl(pickedTitle)})
```

Comment:
Will create a ticket, I think we should validate releases names are not repeated

Reply:
definitely ✅